### ISSUE TITLE
Written tutorials

### ISSUE THREAD

**Issue by OyaCanli** [enpassio/Databinding#15]
I open this to discuss the form of written tutorials.
-Shall we make in parts? A seperate tutorial for each project?
-Shall we do seperate tutorials for Java and Kotlin versions? Or a common article which contains snippets from both?
-Do we need to cover it like a lesson, or rather like a project walkthrough?

**Comment by rajtheinnovator** [comment]
Oya, the basic idea is that we let the developers learn the concept first
but then also help them how to convert that concept in some real use
through a project walk through. So the step/strategy I could think here is
that, we can create a series which in the beginning, helps developers learn
main parts and concepts using just a part of the related code snippet. Then
in the later parts of the series, we can walk them through whole of the
project. This is the things which is missing there on web. We've lots of
tutorials which discuss the main parts of the concepts but it's hard to use
them in a project because of a missing definite structure/steps.

Regarding Java/Kotlin, I think we can do just one language and when giving
the code, we can give code from one language and for the code for other
language, we can share its link from the related branch.

**Comment by OyaCanli** [comment]
@rajtheinnovator For the article, I thought of putting small snippets as codeblocks and larger snippets as gists. I guess there is no need to create gists for a few lines of code. What do you think?

And you proposed to create seperate gists for kotlin and java versions. Then which one will we embed in the article? Will we add both versions? Or embed kotlin, and share the link for java, for instance?

**Comment by rajtheinnovator** [comment]
@OyaCanli yeah, that'd be good. We can keep small blocks in the article itself instead of using Gists, but we should take care of code formatting. As I've not written any medium article so I'm not sure how good it handles code formatting. Hopefully, it should handle it well as Medium is now quite an improved platform.

If you're going to talk about Kotlin part mostly, then we can embed Kotlin code, similarly for Java. The language which is not being talked directly in the article, the related code can be put via a link
